### ISSUE TITLE
Document outline bug with HTML symbols

### ISSUE THREAD

**Issue by bcmadsen** [RickStrahl/MarkdownMonster#531]
Open a new document. Enter this text:

```text
# Prologue
Hello
# Voil&agrave;!
Have a beer!!
# Voil&agrave; encore!
Time for another beer!
# Tsch&uuml;&szlig;
Good-bye
```

Now ask MM to add a document outline, which gives us this:

```text
<!-- Start Document Outline -->

* [Prologue](#prologue)
* [Voilà!](#voilà)
* [Voilà encore!](#voilà-encore)
* [Tschüß](#tschüß)

<!-- End Document Outline -->
```

All of this looks exactly like it should in most places -- in the edit window, in the preview window, even in the generated document outline text. Where it doesn't show up like you'd expect is in the left-hand pane's Document Outline:

![image](https://user-images.githubusercontent.com/42981101/59874440-0eb7b580-9353-11e9-8235-5b92aea928c6.png)

I realize that this is a bit of a grey area &ndash; diacritics aren't 7-bit ASCII characters and so I won't be surprised if it turns out that they aren't legal as hyperlink targets &ndash; but they're actually working great as links, even though they contain diacritics &ndash; and so if the left-hand pane could show the diacritics too, we'd be all the way to the goal line, it seems to me, even if this is a grey area of the law.

**Comment by RickStrahl** [comment]
Hmmm... this looks like a bug in Markdig's syntax tree parser:

![image](https://user-images.githubusercontent.com/1374013/59984303-741ed700-95dd-11e9-90cc-013147a97d84.png)

Looks like it's not picking up text following an HTML entity.  I have to dig a little more but I'm guessing it's splitting up the text on that line at the HTML Entity definition. I doubt that's by design but I'll dig around and post an issue with Markdig.

**Comment by RickStrahl** [comment]
Turns out - not a bug in Markdig, but rather a case of a really fragmented document tree with HTML entities showing up as separate elements that have to be combined.

It's fixed now:

![image](https://user-images.githubusercontent.com/1374013/59985031-ac291880-95e3-11e9-8325-5f426c0f91e7.png)

Fixed in: ca0c3930b9fc977fce2efedec1de70a8f53ef3b5
